### ISSUE TITLE
Cross-platform API to get all system font families

### ISSUE THREAD

**Issue by hungnd1475** [unoplatform/uno#5004]
## What would you like to be added:

A cross-platform API that can be used to retrieve all system font families.

## Why is this needed:

Right now there is no easy way for developers to get all system font families across all platforms. This enhancement might be very useful for LOB applications with rich text editing capabilities.

## For which Platform:

- [x] iOS
- [x] Android
- [x] WebAssembly
- [x] WebAssembly renderers for Xamarin.Forms
- [x] macOS
- [x] Skia
  - [x] WPF
  - [x] GTK (Linux)
  - [x] Tizen
- [x] Windows
- [ ] Build tasks
- [ ] Solution Templates

**Comment by carldebilly** [comment]
For Windows, you can use that: http://microsoft.github.io/Win2D/html/M_Microsoft_Graphics_Canvas_Text_CanvasTextFormat_GetSystemFontFamilies.htm (from the [`Win2D.uwp` NuGet package](https://www.nuget.org/packages/Win2D.uwp).

But it's not part of the UWP/WinUI contract, so not available right now on Uno. But...

* On WASM, you can use the [`document.fonts`](https://developer.mozilla.org/en-US/docs/Web/API/Document/fonts) javscript API
* On Android, I'm not sure if there's an API for that. I think there's very few fonts available for apps and you need to supply those you want with your app.
* On iOS 12+, there's the [`UIFont.FamilyNames`](https://docs.microsoft.com/en-us/dotnet/api/uikit.uifont.familynames?view=xamarin-ios-sdk-12)
* For Skia, you can call [`SKFontManager.FontFamilies`](https://docs.microsoft.com/en-us/dotnet/api/skiasharp.skfontmanager.fontfamilies)

**Comment by hungnd1475** [comment]
@carldebilly I am aware of using platform-specific code, but it would be nicer if we have a single API handling this use case. Right now we need to stick with this workaround I guess. Anyway thanks for pointing me to those links, it would save me some time searching for them.

**Comment by carldebilly** [comment]
I agree it would be a good idea to create a unified API for that. But since it's not present in UWP/WinUI, it would need to be in the Uno toolkit.

**Comment by LeftTwixWand** [comment]
For me, System.Drawing.Font is not working on a droid project....

**Comment by c-npolyak** [comment]
Guys, unless I do not understand something, it should not be very difficult to create a platform independent way of getting the fonts. Why has not it been done?

**Comment by jeromelaban** [comment]
@c-npolyak we mostly work by priorities on our customers, and this has not been requested yet. 

That being said, at the time this question was asked, UWP was not providing this and while WinAppSDK does not, [Uno.Toolkit](https://github.com/unoplatform/uno.toolkit.ui) may be good place to provide an API for such a feature that could be available for more platforms, WinAppSDK included.

**Comment by MartinZikmund** [comment]
I think Uno Extensions would be a better place for this - can we move this issue there @jeromelaban ?

**Comment by jeromelaban** [comment]
This looks like a toolkit feature rather than extensions, I'll move it there.
